pkg/gui/presentation/commits.go

```diff
@@ -312,9 +312,31 @@ func displayCommit(
 	bisectInfo *git_commands.BisectInfo,
 	isYouAreHereCommit bool,
 ) []string {
-	hashColor := getHashColor(commit, diffName, cherryPickedCommitHashSet, bisectStatus, bisectInfo)
 	bisectString := getBisectStatusText(bisectStatus, bisectInfo)
 
+	hashString := commit.Hash
+	hashColor := getHashColor(commit, diffName, cherryPickedCommitHashSet, bisectStatus, bisectInfo)
+	hashLength := common.UserConfig.Gui.CommitHashLength
+	if hashLength == 0 && !icons.IsIconEnabled() { // if no icons and no hash, show a char so user still sees status color
+		hashString = hashColor.Sprint(common.UserConfig.Gui.CommitStatusChar)
```

Comment:
I wanted the minimum of commitHashLength to be 0, but then you wouldn't be able to see the status color (if you didn't have icons on), so I added a string that displays in that case.